core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -165,6 +166,11 @@ private[spark] class Executor(
     }
   }
 
+  // Plugins need to load using a class loader that includes the executor's user classpath
+  private val plugins: Option[PluginContainer] = Utils.withContextClassLoader(replClassLoader) {
```

Comment:
I am not sure that plugins should be loaded at this stage when running in local mode, maybe only the driver side of the plugin is sufficient in local mode?
Metrics source registration at this stage, when executed in local mode, will not get the application id. Other metrics handled in executors.scala are not registered when running in local mode.
The current implementation of executor plugins sends a "isLocal boolean" via the pluginContext to handle this case in the plugin logic.

Reply:
You can check if you're running in local mode by looking at the spark.master value. I intentionally did not add to the API since it would be redundant.

I'm also not especially worried about local mode. It's mostly for debugging. If something doesn't work 100% as intended for plugins, I'm totally fine with it.

Reply:
Indeed, that should be fine.